internal/server/create_document.go

```diff
@@ -0,0 +1,293 @@
+				return err
+			}
+
+			data.DocumentTemplateTypes = translateDocumentData(data.DocumentTemplates, data.DocumentTemplateRefData)
+
+			templateId := r.FormValue("templateId")
+			if templateId != "" {
+				for _, dt := range data.DocumentTemplates {
+					if dt.TemplateId == templateId {
+						data.TemplateSelected = dt
+						break
+					}
+				}
+			}
+			if data.TemplateSelected.TemplateId != "" {
+				data.DocumentInsertTypes = translateInsertData(data.TemplateSelected.Inserts, data.DocumentTemplateRefData)
+			}
+
+			hasViewedInsertPage := r.FormValue("hasViewedInserts")
+			if hasViewedInsertPage == "true" {
+				data.HasViewedInsertPage = true
+				data.SelectedInserts = r.Form["insert"]
+				data.Recipients, err = getRecipients(ctx, client, data.Case)
+				if err != nil {
+					return err
+				}
+			}
+
+		case http.MethodPost:
+			recipientControls := postFormString(r, "recipientControls")
+
+			switch recipientControls {
+			case "select":
+				templateId := r.FormValue("templateId")
+				inserts := r.Form["insert"]
+				if len(inserts) == 0 {
+					inserts = []string{}
+				}
+
+				selectedRecipientIDs, err := sliceAtoi(r.Form["selectRecipients"])
+				if err != nil {
+					return err
+				}
+
+				for _, recipientID := range selectedRecipientIDs {
+					_, err = client.CreateDocument(ctx, caseID, recipientID, templateId, inserts)
+					if err != nil {
+						return err
+					}
+				}
+
+				if ve, ok := err.(sirius.ValidationError); ok {
+					w.WriteHeader(http.StatusBadRequest)
+					data.Error = ve
+				} else if err != nil {
+					return err
+				} else {
+					data.Success = true
+					// redirect
+				}
+
+			case "generate":
+				contact := sirius.Person{
+					Salutation:            postFormString(r, "salutation"),
+					Firstname:             postFormString(r, "firstname"),
+					Middlenames:           postFormString(r, "middlenames"),
+					Surname:               postFormString(r, "surname"),
+					CompanyName:           postFormString(r, "companyName"),
+					CompanyReference:      postFormString(r, "companyReference"),
+					AddressLine1:          postFormString(r, "addressLine1"),
+					AddressLine2:          postFormString(r, "addressLine2"),
+					AddressLine3:          postFormString(r, "addressLine3"),
+					Town:                  postFormString(r, "town"),
+					County:                postFormString(r, "county"),
+					Postcode:              postFormString(r, "postcode"),
+					IsAirmailRequired:     postFormString(r, "isAirmailRequired") == "Yes",
+					PhoneNumber:           postFormString(r, "phoneNumber"),
+					Email:                 postFormString(r, "email"),
+					CorrespondenceByPost:  postFormCheckboxChecked(r, "correspondenceBy", "post"),
+					CorrespondenceByEmail: postFormCheckboxChecked(r, "correspondenceBy", "email"),
+					CorrespondenceByPhone: postFormCheckboxChecked(r, "correspondenceBy", "phone"),
+					CorrespondenceByWelsh: postFormCheckboxChecked(r, "correspondenceBy", "welsh"),
```

Comment:
I know this is matching what we have, but I think these aren't needed because they won't affect the letter (and non-case contacts aren't used for anything else)

Reply:
I see what you are saying but i'm a bit hesitant to change how the form currently works. I can check with katie and see what she thinks